user-interface/1004-ASSO-associate.md

```diff
@@ -1 +1,69 @@
+
+The word "associate" is used in some user interfaces, as apposed the word "stake" in function names. Stake can be avoided to prevent users thinking they would get a return only after the staking step. On Vega `Staking = Association + Nomination`, as in you need to run the "stake " function on Ethereum but then the nominate step on Vega before you get staking income. See [Glossary](../glossaries/staking-and-governance.md).
+
+Associated tokens also count as vote weight in on-Vega governance (new markets etc), A parties vote weight is backed by the number of Governance tokens associated with that Vega key/party.
+
+Associating tokens to a Vega key is a little like depositing, except, a deposit can only be released by the Vega network, where as an association can be revoked on ethereum by the the eth key that did the association.
+
+Governance tokens may be held by a [Vesting contract](1005-VEST-vesting.md).
+
+## Token discovery
+
+When looking to acquire governance tokens, I...
+
+- **must** see the contract address for the governance token of the Vega network <a name="1004-ASSO-001" href="#1004-ASSO-001">1004-ASSO-001</a>
+
+...so I can participate in governance and staking.
+
+## Associate
+
+When looking to stake validators or participate in governance, I first need to associate governance tokens with a Vega wallet/key, I...
+
+- **must** [connect an Ethereum wallet/key](0004-EWAL-connect_ethereum_wallet.md) to see tokens it may have in wallet or attributed to it in the vesting contract <a name="1004-ASSO-002" href="#1004-ASSO-002">1004-ASSO-002</a>
+- **must** select a Vega key to associate to <a name="1004-ASSO-003" href="#1004-ASSO-003">1004-ASSO-003</a>
+  - **must** be able use a [connected Vega wallet](0002-WCON-connect_vega_wallet.md) as instead of manually inputting a public key <a name="1004-ASSO-004" href="#1004-ASSO-004">1004-ASSO-004</a>
+  - **should** be able to populate field with a string, so I can associate to a wallet I without connecting it
+- if the connected ethereum wallet has vesting tokens: **must** be able to select to associate from either the vesting contract or the wallet <a name="1004-ASSO-006" href="#1004-ASSO-006">1004-ASSO-006</a>
+- **must** see the number of un-associated tokens in the selected wallet/vesting contract <a name="1004-ASSO-007" href="#1004-ASSO-007">1004-ASSO-007</a>
+- **must** select the amount of tokens to associate <a name="1004-ASSO-008" href="#1004-ASSO-008">1004-ASSO-008</a>
+  - **must** be able to populate the input with the amount of un-associated tokens for the selected wallet/vesting contract <a name="1004-ASSO-009" href="#1004-ASSO-009">1004-ASSO-009</a>
+- **must** be warned if the amount being associated is greater than the amount available in the connected ethereum wallet <a name="1004-ASSO-010" href="#1004-ASSO-010">1004-ASSO-010</a>
```

Comment:
warned or prevented from submitting?

Reply:
Yeah, as a general rule I've gone for warned rather than prevented. A specific app could choose to prevent it happening too.
It mostly came up with submitting orders (where the trading mode is not applicable for what your ticket is armed with but there are cases you might want to submit anyway).

What do you think?
In this case meta mask should show an exception error right?